Source/JavaScriptCore/assembler/MacroAssemblerX86_64.h

```diff
@@ -964,8 +964,16 @@ class MacroAssemblerX86_64 : public MacroAssemblerX86Common {
 
     void sub64(RegisterID a, TrustedImm32 imm, RegisterID dest)
     {
-        move(a, dest);
-        sub64(imm, dest);
+        if (src == dest) {
+            sub64(imm, dest);
+            return;
+        }
+
+        if (UNLIKELY(imm.m_value == INT32_MIN)) {
```

Comment:
Why not have a similar check for sub32?

Reply:
For `sub32`, we can just pass the `INT32_MIN` as a `TrustedImm32` - because it's a 32-bit operation, that's valid, because `-INT32_MIN == INT32_MIN` for 32-bit integers. For this function though, `-INT32_MIN` _is_ representable as a positive integer in the 64-bit value space, so it's not correct for us to truncate it to a 32-bit immediate in `lea`.